processors/src/MCTrackerHitProcessor.cxx

```diff
@@ -78,7 +78,13 @@ bool MCTrackerHitProcessor::process(IEvent* ievent) {
         hitPos[2] = lcio_mcTracker_hit->getPosition()[2];
         mc_tracker_hit->setPosition(hitPos);
 
-        // Set the energy deposit of the hit
+	double hitMom[3];
+	hitMom[0] = lcio_mcTracker_hit->getMomentum()[0];
+	hitMom[1] = lcio_mcTracker_hit->getMomentum()[1];
+	hitMom[2] = lcio_mcTracker_hit->getMomentum()[2];
+	mc_tracker_hit->setMomentum(hitMom);
```

Comment:
Is there a reason you didn't have `setMomentum` take in the type returned by `getMomentum()`?
e.g.
```suggestion
mc_tracker_hit->setMomentum(lcio_mcTracker_hit->getMomentum())
```
That just feels cleaner and I don't know why you couldn't do that. I think you'd just want to change the `double` to `float` and then it would work.

Reply:
@tomeichlersmith 
I've followed the suggestion. Unfortunately I can't harmonize `setPosition` with `setMomentum` as one needs `double` and the other `float` to match LCIO signature.